cpp/src/parquet/column_writer.cc

```diff
@@ -249,9 +249,9 @@ int LevelEncoder::Encode(int batch_size, const int16_t* levels) {
 class SerializedPageWriter : public PageWriter {
  public:
   SerializedPageWriter(std::shared_ptr<ArrowOutputStream> sink, Compression::type codec,
-                       int compression_level, ColumnChunkMetaDataBuilder* metadata,
-                       int16_t row_group_ordinal, int16_t column_chunk_ordinal,
-                       bool use_page_checksum_verification,
+                       const std::shared_ptr<CodecOptions>& codec_options,
```

Comment:
No need for `shared_ptr` here.
```suggestion
                       const CodecOptions& codec_options,
```